Assets/_BForBoss/_Character/Scripts/PlayerDashBehaviour.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using ECM2.Characters;
+using ECM2.Components;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace BForBoss
+{
+    public class PlayerDashBehaviour : MonoBehaviour
+    {
+        [SerializeField] private float _dashImpulse = 20.0f;
+        [SerializeField] private float _dashDuration = 0.2f;
+        [SerializeField] private float _dashCoolDown = 0.5f;
+
+        private float _dashElapsedTime = 0;
+        private float _dashCoolDownElapsedTime = 0;
+        private bool _isDashing = false;
+        private Vector3 _dashingDirection = Vector3.zero;
+        private Func<Vector2> _characterInputMovement = null;
+
+        private Character _baseCharacter = null;
+
+        private bool IsCoolDownOver => _dashCoolDownElapsedTime <= 0;
+        
+        public void Initialize(Character baseCharacter, Func<Vector2> characterMovement)
+        {
+            _baseCharacter = baseCharacter;
+            _characterInputMovement = characterMovement;
+        }
+        
+        public void SetupPlayerInput(InputAction dashAction)
+        {
+            dashAction.started += OnDash;
+            dashAction.canceled += OnDash;
+            dashAction.Enable();
```

Comment:
I think you might want to be also disabling this in on disable, I'm not sure how important it is but I'm used to seeing that in input action code